src/shared/exec-util.c

```diff
@@ -93,6 +93,8 @@ static int do_execute(
         bool parallel_execution;
         int r;
 
+        assert_se(unsetenv("NOTIFY_SOCKET") == 0);
```

Comment:
maybe add a comment explaining why it's done

Reply:
That sentence is very awkward. In particular, negation is missing from the last part…